source/functions/SimulinkModelFunctions/regnLYaw.m

```diff
@@ -0,0 +1,69 @@
+function [Fext,relYawLast,coeffsLastMD,coeffsLastRE,coeffsLastIM] = regnLYaw(A,w,dofGRD,dirGRD,fEHRE,fEHIM, fEHMD,time,WaveDir,Disp, intThresh, prevYaw, prevCoeffMD, prevCoeffRE, prevCoeffIM)
+%#codegen
+% Fext is the excitation force output, relYawLast is the angle of relative
+% yaw at which that last interpolation was performed. If the current
+% relative yaw angle - last interpolated > threshold, the interpolation is
+% performed again. To interpolate every time step, let threshold=0. 
+
```

Comment:
The Simulink block `regNLYaw` also has some parameters initialized, is that intended?

```matlab

function [Fext,relYawLast,coeffsLastMD,coeffsLastRE,coeffsLastIM] = regnLYaw(A,w,dofGRD,dirGRD,fEHRE,fEHIM, fEHMD,time,WaveDir,Disp, intThresh, prevYaw, prevCoeffMD, prevCoeffRE, prevCoeffIM)
%#codegen
% Fext is the excitation force output, relYawLast is the angle of relative
% yaw at which that last interpolation was performed. If the current
% relative yaw angle - last interpolated > threshold, the interpolation is
% performed again. To interpolate every time step, let threshold=0. 

%initialize outputs
Fext = zeros(1,6);
relYawLast=0;
coeffsLastMD=zeros(length(w),6); % should be (1,6) for regular waves
coeffsLastRE=zeros(length(w),6);
coeffsLastIM=zeros(length(w),6);

[Fext,relYawLast,coeffsLastMD,coeffsLastRE,coeffsLastIM] = regnLYaw(A,w,dofGRD,dirGRD,fEHRE,fEHIM, fEHMD,time,WaveDir,Disp, intThresh, prevYaw, prevCoeffMD, prevCoeffRE, prevCoeffIM);

end

```

Reply:
After talking with Dominic, it seems the nLYaw cases need to be initialized within the matlab function from Simulink due to the nature of the function. Initialization within the m-file or Simulink led to errors.

Reply:
Same as above, added to project board